packages/generator-langium/templates/vscode/esbuild.mjs

```diff
@@ -0,0 +1,54 @@
+//@ts-check
+import * as esbuild from 'esbuild';
+
+const watch = process.argv.includes('--watch');
+const minify = process.argv.includes('--minify');
+
+const success = watch ? 'Watch build succeeded' : 'Build succeeded';
+
+function getTime() {
+    const date = new Date();
+    return `[${`${padZeroes(date.getHours())}:${padZeroes(date.getMinutes())}:${padZeroes(date.getSeconds())}`}] `;
+}
+
+function padZeroes(i) {
+    return i.toString().padStart(2, '0');
+}
+
+const plugins = [{
+    name: 'watch-plugin',
+    setup(build) {
+        build.onEnd(result => {
+            if (result.errors.length === 0) {
+                console.log(getTime() + success);
+            }
+        });
+    },
+}];
+
+const ctx = await esbuild.context({
+    // Entry points for the vscode extension and the language server
+    entryPoints: ['src/extension/main.ts', 'src/language/main.ts'],
+    outdir: 'out',
+    bundle: true,
+    target: "es6",
+    // VSCode's extension host is still using cjs, so we need to transform the code
+    format: 'cjs',
+    // To prevent confusing node, we explicitly use the `.cjs` extension
+    outExtension: {
+        '.js': '.cjs'
+    },
+    loader: { '.ts': 'ts' },
+    external: ['vscode'],
+    platform: 'node',
+    sourcemap: !minify,
+    minify,
+    plugins
+});
+
+if (watch) {
+    await ctx.watch();
+} else {
+    await ctx.rebuild();
```

Comment:
Minor thing...

Use a try-finally here, insert the if-then-else without dispose and for the finally block insert the dispose...

One could argue to remove the dispose because the process will be dead so or so.

Reply:
I don't think that esbuild throws. It will just callback with the error. And even if it were to throw, then the process would end either way, making it dispose all resources. Using try-finally here would only make things more complicated.

Reply:
I have a very splitted opinion here. It is actually ignorable, because it really makes no sense due to the end of the lifetime of the process.

For the watch branch we say: It will never throw, so cleanup is when process is killed.
For the rebuild branch we dispose, because there is a no endless loop. So it makes sense to cleanup. But if you step out the process is cleanup nevertheless...

You do not need to act here, it is just a weird situation, you can have four different scenarios (no clean up in both branches, cleanup in both branches, cleanup only in one branch x 2), but they all will lead to the same result... meh

Reply:
> For the rebuild branch we dispose, because there is a no endless loop. So it makes sense to cleanup. But if you step out the process is cleanup nevertheless...

Note that we cleanup because `ctx.rebuild()` does something in the background which prevents the process from ending. Calling `dispose` ends that background task and allows the process to end. If the `rebuild` were to throw (which I've never experienced, since an erroneous build is a part of the "happy path"), the process would just crash.